router/core/operation_processor.go

```diff
@@ -308,6 +327,13 @@ func (o *OperationKit) unmarshalOperation() error {
 			}
 		}
 		if o.parsedOperation.GraphQLRequestExtensions.PersistedQuery != nil {
+			if !o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.isValidHash() {
+				return &httpGraphqlError{
+					message:    "persistedQuery.sha256Hash must be a valid sha256 hash",
```

Comment:
`persistedQuery.` is internal. Not sure, how meaningful this error message is to the user. Maybe a more fluent error description would be better here.

Reply:
Please see https://github.com/wundergraph/cosmo/pull/2078#discussion_r2235644495 . All errors in the `unmarshalOperation` are rewritten by the callee. I am not brave enough to expose them all and break so many tests. :-)

Reply:
Anyway, I have changed the error message to be more human friendly: "persisted query does not have a valid sha256 hash"

Reply:
@StarpTech  I have fixed the errors handling, please check it out.